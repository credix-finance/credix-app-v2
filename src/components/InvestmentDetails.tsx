import React from "react";
import { InvestmentReturn } from "@components/InvestmentReturn";

interface InvestmentDetailsProps {
	balance: number;
	balanceCurrency: string;
	investments: number;
	investmentsCurrency: string;
	investmentsReturn: number;
}

export const InvestmentDetails = ({
	balance,
	balanceCurrency,
	investments,
	investmentsCurrency,
	investmentsReturn,
}: InvestmentDetailsProps) => {
	return (
		<div className="md:flex space-y-2 md:space-y-0 md:space-x-4">
			<div className="rounded border border-solid border-neutral-40 bg-neutral-0 p-6 grid place-items-center md:block">
				<div className="text-xs md:text-base font-medium">Current balance</div>
				<div className="text-2xl font-bold">{`${balance} ${balanceCurrency}`}</div>
			</div>
			<div className="rounded border border-solid border-neutral-40 bg-neutral-0 p-6 grid place-items-center md:block">
				<div className="flex justify-between items-start md:space-x-24 md:justify-start">
					<div className="text-xs md:text-base font-medium">Current investments</div>
					<div className="hidden md:block">
						<InvestmentReturn value={investmentsReturn} />
					</div>
				</div>
				<div>
					<div className="text-2xl font-bold">{`${investments} ${investmentsCurrency}`}</div>
				</div>
			</div>
		</div>
	);
};
