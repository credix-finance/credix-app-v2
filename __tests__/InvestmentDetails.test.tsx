import React from "react";
import renderer from "react-test-renderer";
import { InvestmentDetails } from "@components/InvestmentDetails";
import { TokenAmount } from "@solana/web3.js";
import Big from "big.js";

test("Positive return", () => {
	const props = {
		balance: {
			uiAmountString: "65",
		} as TokenAmount,
		balanceCurrency: "USDC",
		investments: new Big(256),
		investmentsCurrency: "USDC",
		investmentsReturn: 3.43,
	};

	const component = renderer.create(<InvestmentDetails {...props} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test("Negative return", () => {
	const props = {
		balance: {
			uiAmountString: "65",
		} as TokenAmount,
		balanceCurrency: "USDC",
		investments: new Big(256),
		investmentsCurrency: "USDC",
		investmentsReturn: -3.43,
	};

	const component = renderer.create(<InvestmentDetails {...props} />);

	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
