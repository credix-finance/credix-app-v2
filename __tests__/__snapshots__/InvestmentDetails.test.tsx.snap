// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Negative return 1`] = `
<div
  className="md:flex space-y-2 md:space-y-0 md:space-x-4"
>
  <div
    className="rounded border border-solid border-neutral-40 bg-neutral-0 p-6 grid place-items-center md:block"
  >
    <div
      className="text-xs md:text-base font-medium"
    >
      Current balance
    </div>
    <div
      className="text-2xl font-bold"
    >
      65 USDC
    </div>
  </div>
  <div
    className="rounded border border-solid border-neutral-40 bg-neutral-0 p-6 grid place-items-center md:block"
  >
    <div
      className="flex justify-between items-start md:space-x-24 md:justify-start"
    >
      <div
        className="text-xs md:text-base font-medium"
      >
        Current investments
      </div>
      <div
        className="flex items-center space-x-[10px] hidden md:flex"
      >
        <svg
          className="w-4 h-4 fill-current text-action-hover w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="currentColor"
            height="24"
            rx="12"
            width="24"
          />
          <path
            d="M18.6139 15.5867C18.5463 15.7496 18.4168 15.879 18.2539 15.9467C18.1738 15.9808 18.0877 15.9989 18.0006 16H14.6673C14.4905 16 14.3209 15.9298 14.1959 15.8047C14.0708 15.6797 14.0006 15.5101 14.0006 15.3333C14.0006 15.1565 14.0708 14.987 14.1959 14.8619C14.3209 14.7369 14.4905 14.6667 14.6673 14.6667H16.3939L12.6673 10.94L10.4739 13.14C10.412 13.2025 10.3382 13.2521 10.257 13.2859C10.1758 13.3198 10.0886 13.3372 10.0006 13.3372C9.9126 13.3372 9.82547 13.3198 9.74423 13.2859C9.66299 13.2521 9.58925 13.2025 9.52728 13.14L5.52728 9.14C5.46479 9.07802 5.4152 9.00429 5.38135 8.92305C5.3475 8.84181 5.33008 8.75467 5.33008 8.66667C5.33008 8.57866 5.3475 8.49152 5.38135 8.41028C5.4152 8.32904 5.46479 8.25531 5.52728 8.19333C5.58925 8.13085 5.66299 8.08125 5.74423 8.04741C5.82547 8.01356 5.9126 7.99613 6.00061 7.99613C6.08862 7.99613 6.17576 8.01356 6.25699 8.04741C6.33823 8.08125 6.41197 8.13085 6.47394 8.19333L10.0006 11.7267L12.1939 9.52667C12.2559 9.46418 12.3297 9.41459 12.4109 9.38074C12.4921 9.34689 12.5793 9.32947 12.6673 9.32947C12.7553 9.32947 12.8424 9.34689 12.9237 9.38074C13.0049 9.41459 13.0786 9.46418 13.1406 9.52667L17.3339 13.7267V12C17.3339 11.8232 17.4042 11.6536 17.5292 11.5286C17.6542 11.4036 17.8238 11.3333 18.0006 11.3333C18.1774 11.3333 18.347 11.4036 18.472 11.5286C18.597 11.6536 18.6673 11.8232 18.6673 12V15.3333C18.6662 15.4205 18.6481 15.5065 18.6139 15.5867Z"
            fill="white"
          />
        </svg>
        <div
          className="text-action-hover font-bold"
        >
          -3.43
          %
        </div>
      </div>
    </div>
    <div>
      <div
        className="text-2xl font-bold"
      >
        256 USDC
      </div>
    </div>
  </div>
</div>
`;

exports[`Positive return 1`] = `
<div
  className="md:flex space-y-2 md:space-y-0 md:space-x-4"
>
  <div
    className="rounded border border-solid border-neutral-40 bg-neutral-0 p-6 grid place-items-center md:block"
  >
    <div
      className="text-xs md:text-base font-medium"
    >
      Current balance
    </div>
    <div
      className="text-2xl font-bold"
    >
      65 USDC
    </div>
  </div>
  <div
    className="rounded border border-solid border-neutral-40 bg-neutral-0 p-6 grid place-items-center md:block"
  >
    <div
      className="flex justify-between items-start md:space-x-24 md:justify-start"
    >
      <div
        className="text-xs md:text-base font-medium"
      >
        Current investments
      </div>
      <div
        className="flex items-center space-x-[10px] hidden md:flex"
      >
        <svg
          className="w-4 h-4 fill-current text-action-primary w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="currentColor"
            height="24"
            rx="12"
            width="24"
          />
          <path
            d="M18.6139 8.41333C18.5463 8.25043 18.4168 8.12098 18.2539 8.05333C18.1738 8.01917 18.0877 8.00105 18.0006 8H14.6673C14.4905 8 14.3209 8.07024 14.1959 8.19526C14.0708 8.32029 14.0006 8.48986 14.0006 8.66667C14.0006 8.84348 14.0708 9.01305 14.1959 9.13807C14.3209 9.2631 14.4905 9.33333 14.6673 9.33333H16.3939L12.6673 13.06L10.4739 10.86C10.412 10.7975 10.3382 10.7479 10.257 10.7141C10.1758 10.6802 10.0886 10.6628 10.0006 10.6628C9.9126 10.6628 9.82547 10.6802 9.74423 10.7141C9.66299 10.7479 9.58925 10.7975 9.52728 10.86L5.52728 14.86C5.46479 14.922 5.4152 14.9957 5.38135 15.0769C5.3475 15.1582 5.33008 15.2453 5.33008 15.3333C5.33008 15.4213 5.3475 15.5085 5.38135 15.5897C5.4152 15.671 5.46479 15.7447 5.52728 15.8067C5.58925 15.8692 5.66299 15.9187 5.74423 15.9526C5.82547 15.9864 5.9126 16.0039 6.00061 16.0039C6.08862 16.0039 6.17576 15.9864 6.25699 15.9526C6.33823 15.9187 6.41197 15.8692 6.47394 15.8067L10.0006 12.2733L12.1939 14.4733C12.2559 14.5358 12.3297 14.5854 12.4109 14.6193C12.4921 14.6531 12.5793 14.6705 12.6673 14.6705C12.7553 14.6705 12.8424 14.6531 12.9237 14.6193C13.0049 14.5854 13.0786 14.5358 13.1406 14.4733L17.3339 10.2733V12C17.3339 12.1768 17.4042 12.3464 17.5292 12.4714C17.6542 12.5964 17.8238 12.6667 18.0006 12.6667C18.1774 12.6667 18.347 12.5964 18.472 12.4714C18.597 12.3464 18.6673 12.1768 18.6673 12V8.66667C18.6662 8.57955 18.6481 8.49348 18.6139 8.41333Z"
            fill="white"
          />
        </svg>
        <div
          className="text-action-primary font-bold"
        >
          +3.43
          %
        </div>
      </div>
    </div>
    <div>
      <div
        className="text-2xl font-bold"
      >
        256 USDC
      </div>
    </div>
  </div>
</div>
`;
